lib/msf/core/auxiliary/ubiquiti.rb

```diff
@@ -0,0 +1,336 @@
+    def process_device(lines)
+      lines.each do |line|
+        report_host({
+         :host => line['ip'],
+         :name => line['name'],
+         :mac => line['mac'],
+         :os_name => 'Ubiquiti Unifi'
+        })
+        print_good("Unifi Device #{line['name']} of model #{line['model']} on #{line['ip']}")
+      end
+    end
+
+    # Example lines
+    # {"__cmd"=>"select", "collection"=>"user"}
+    # {"_id"=>BSON::ObjectId('3c3e21ac3715ce20a721d9ba'), "mac" => "00:0c:29:11:aa:11", "site_id" => "5c7f111b1111aa2087d11111", "oui" => "Vmware", "is_guest" => false, "first_seen" => 1551111161, "last_seen" => 1561621747, "is_wired" => true, "hostname" => "android", "usergroup_id" => "", "name" => "example device", "noted" => true, "use_fixedip" =>  true, "network_id" => "1c7f111a1115aa2087aaa9aa", "fixed_ip" => "7.7.7.7"}
+    def process_user_device(lines)
+      lines.each do |line|
+        host_hash = {
+         :name => line['hostname'],
+         :mac => line['mac']
+        }
+        desc = "#{line['hostname']} (#{line['mac']})"
+        if line['fixed_ip'] 
+          host_hash[:host] = line['fixed_ip']
+          desc << " on IP #{line['fixed_ip']}"
+        end
+        if line['name'] 
+          host_hash[:info] = line['name']
+          desc << " with name #{line['name']}"
+        end
+        report_host(host_hash)
+        print_good("Network Device #{desc} found")
+      end
+    end
+
+    # here is where we actually process the file
+    config.each do |key,value|
+      case key
+      when 'admin'
+        process_admin(value, credential_data)
+      when 'radiusprofile'
+        process_radiusprofile(value, credential_data)
+      when 'setting'
+        process_settings(value, credential_data)
+      when 'firewallrule'
+        process_firewallrule(value)
+      when 'wlanconf'
+        process_wlanconf(value, credential_data)
+      when 'firewallgroup'
+        process_firewallgroup(value)
+      when 'device'
+        process_device(value)
+      when 'user'
+        process_user_device(value)
+      end
```

Comment:
You could make this quite a bit more dynamic by using respond_to and send here.
The whole case block could reduced to
```ruby
next unless self.respond_to? "process_#{key}"
self.send("process_#{key}", value)
```

`process_user_device` would need to be switched to `process_user` and you'd need an alternative way to ingest the `credential_data` hash, either a function would work (you wouldn't have to `.dup` it any more) or using a constant `CREDENTIAL_TEMPLATE`.

Reply:
I'm not opposed to cleaning this up, but this is based off of the cisco module. Brocade, Juniper, and Unifi are all based on the Cisco module, so I'd rather hold off on this change and do a diff PR which keeps all of those in line and upgrades them all at once.